Revit_Core_Engine/Convert/Environment/ToRevit/ElementType.cs

```diff
@@ -0,0 +1,191 @@
+        {
+            if (property == null || document == null)
+                return null;
+
+            Autodesk.Revit.DB.Electrical.WireType elementType = refObjects.GetValue<Autodesk.Revit.DB.Electrical.WireType>(document, property.BHoM_Guid);
+            if (elementType != null)
+                return elementType;
+
+            settings = settings.DefaultIfNull();
+
+            elementType = property.ElementType(document, categories, settings) as Autodesk.Revit.DB.Electrical.WireType;
+            if (elementType != null)
+                return elementType;
+
+            List<Autodesk.Revit.DB.Electrical.WireType> wireTypes = new FilteredElementCollector(document).OfClass(typeof(Autodesk.Revit.DB.Electrical.WireType)).Cast<Autodesk.Revit.DB.Electrical.WireType>().ToList();
+
+            if (property.SectionProfile.ElementProfile is BH.oM.Spatial.ShapeProfiles.TubeProfile)
+                elementType = wireTypes.FirstOrDefault(x => x.FamilyName == "Round Wire"); // ToDo: Check this.
+
+            if (elementType != null)
+            {
+                BH.Engine.Reflection.Compute.RecordNote("Wire is being pushed as the first type available in the Revit model.");
+            }
+            else
+                return null;
+
+            // Copy parameters from BHoM object to Revit element
+            elementType.CopyParameters(property, settings);
+
+            refObjects.AddOrReplace(property, elementType);
+            return elementType;
+        }
+
+        /***************************************************/
+
+        // Todo Check FamilyName == Default @ 173
+
+        public static Autodesk.Revit.DB.Electrical.CableTrayType ToRevitElementType(this oM.MEP.System.SectionProperties.CableTraySectionProperty property, Document document, IEnumerable<BuiltInCategory> categories = null, RevitSettings settings = null, Dictionary<Guid, List<int>> refObjects = null)
+        {
+            if (property == null || document == null)
+                return null;
+
+            Autodesk.Revit.DB.Electrical.CableTrayType elementType = refObjects.GetValue<Autodesk.Revit.DB.Electrical.CableTrayType>(document, property.BHoM_Guid);
+            if (elementType != null)
+                return elementType;
+
+            settings = settings.DefaultIfNull();
+
+            elementType = property.ElementType(document, categories, settings) as Autodesk.Revit.DB.Electrical.CableTrayType;
+            if (elementType != null)
+                return elementType;
+
+            List<Autodesk.Revit.DB.Electrical.CableTrayType> trayTypes = new FilteredElementCollector(document).OfClass(typeof(Autodesk.Revit.DB.Electrical.CableTrayType)).Cast<Autodesk.Revit.DB.Electrical.CableTrayType>().ToList();
+
+            if (property.SectionProfile.ElementProfile is BH.oM.Spatial.ShapeProfiles.TubeProfile)
+                elementType = trayTypes.FirstOrDefault(x => x.FamilyName == "Default"); // ToDo: Check this.
```

Comment:
then check :)

Reply:
changing cableTray sectionProfile to box as well as updating Family default name to "Cable Tray with Fittings"